lib/auth/methods.go

```diff
@@ -341,7 +349,11 @@ func (a *Server) authenticateUser(
 }
 
 // Do not use this method directly, use authenticateUser instead.
-func (a *Server) authenticateUserInternal(ctx context.Context, req AuthenticateUserRequest) (mfaDev *types.MFADevice, user string, err error) {
+func (a *Server) authenticateUserInternal(
+	ctx context.Context,
+	req AuthenticateUserRequest,
+	requiredScope mfav1.ChallengeScope,
+) (mfaDev *types.MFADevice, user string, err error) {
 	if err := req.CheckAndSetDefaults(); err != nil {
 		return nil, "", trace.Wrap(err)
 	}
```

Comment:
Could we double-check the scope in the passwordless branch and error if it's wrong?

Reply:
It's already there, in the `authenticatePasswordless` function that doesn't even take the required extensions/scope and just substitutes its own. I can double check here, though.

Reply:
Ok, hold on. That doesn't really make any sense to double-check a parameter that ends up not getting used. @codingllama  Do you mean to pass the required extensions down to `authenticatePasswordless`, and validate it there instead of creating a new one?

Reply:
I meant something like this:

```go
	// Only one path if passwordless, other variants shouldn't see an empty user.
	if passwordless {
		if requiredScope != mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN {
			return nil, "", trace.BadParameter("passwordless can only be used with the PASSWORDLESS scope")
		}
		return a.authenticatePasswordless(ctx, req)
	}
```

This is just so we flag any eventual programming errors or mismatch.

Reply:
Ping on this one? I think it's only comment left.

Reply:
Strange. It was there. Perhaps I accidentally hit the undid it.